### PR TITLE
Fix miscalculation of cjk characters.

### PR DESCRIPTION
**Pull request by bcho**
Patch is taken from #8, and the tests had been fixed.

### CODE REVIEW

File: terminaltables.py [hunk 1]
```diff
@@ -22,21 +22,56 @@
...
+def _get_width(string):
+    """Get the real width of unicode string.
+
+    Positional arguments:
+    string -- unicode string.
+
+    Returns:
+    String width.
+    """
+    if isinstance(string, Color):
```
Comment (Robpol86) [inline terminaltables.py]: Can you do this without isinstance? I don't like isinstance, it makes code inflexible and hard to use in other libraries. What if someone made something like colorclass where they subclass string/unicode? terminaltables wouldn't work with their library.

Comment (bcho) [inline terminaltables.py]: Yes, `isinstance` should be avoid here, but can `Color` provide a interface just like the normal `str`? I mean, when iterating / decoding the `Color` instance, it should return the real value instead of the escaped one, I think this will make things much more easier.

Comment (Robpol86) [inline terminaltables.py]: I think it won't be so easy since something like sys.stdout.write() iterates the Color instance and that expects the color version. You're free to open an issue/pr in colorclass. Maybe instead you can just use hasattr() or getattr() for value_no_colors?

File: terminaltables.py [hunk 2]
```diff
@@ -22,21 +22,56 @@
...
-from colorclass import _WindowsCSBI
+from colorclass import _WindowsCSBI, Color
 
 __author__ = '@Robpol86'
 __license__ = 'MIT'
 __version__ = '1.2.0'
 
 
+if PY2:  # pragma: no cover
+    def _to_unicode(string):
```
Comment (Robpol86) [inline terminaltables.py]: I also don't like this implementation. _to_unicode() is only used in _get_width(). Can you move this block of code into that function?

File: terminaltables.py [hunk 3]
```diff
@@ -22,21 +22,56 @@
...
 import sys
+import unicodedata
+
+PY2 = sys.version_info[0] == 2
 
 try:
     import fcntl
     import termios
-except ImportError:
+except ImportError:  # pragma: no cover
```
Comment (Robpol86) [inline terminaltables.py]: Why is this needed?

File: terminaltables.py [hunk 4]
```diff
@@ -22,21 +22,56 @@
 import re
 import struct
 import sys
+import unicodedata
+
+PY2 = sys.version_info[0] == 2
```
Comment (Robpol86) [inline terminaltables.py]: This is only used (eventually) in _get_width(). Lets move this in there too. I've been doing try/except elsewhere (see ImportError). Let's stick with that for consistency.

File: terminaltables.py (hunk 2 above)
Comment (bcho) [inline terminaltables.py]: But this will create a new function when calling `_get_width` each time, I think this would make thing slower.

File: terminaltables.py (hunk 3 above)
Comment (bcho) [inline terminaltables.py]: Improve the coverage.

File: terminaltables.py (hunk 2 above)
Comment (Robpol86) [inline terminaltables.py]: What about not creating a function but instead merging the logic into _get_width(). Something like: `string = string.decode('u8') if hasattr(string, 'decode') else string`. _to_unicode is a really simple function, I don't think it needs to be it's own function here.

**Comment by bcho** [comment]
@Robpol86 updated, and I made a rebase for these changes.

File: terminaltables.py (hunk 3 above)
Comment (Robpol86) [inline terminaltables.py]: This is already covered by Windows tests on appveyor. Coverage gets merged in codecov: https://codecov.io/github/Robpol86/terminaltables/terminaltables.py?ref=master no need for this comment.

**Comment by Robpol86** [comment]
Everything else looks good. Once you fix that one line I'll accept. Thanks!

**Comment by bcho** [comment]
@Robpol86 Fixed!

**Comment by Robpol86** [comment]
Released https://pypi.python.org/pypi/terminaltables/1.2.1
